component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ActionResourceImpl.java

```diff
@@ -143,6 +144,18 @@ private Response onError(final Throwable re) {
         if (WebApplicationException.class.isInstance(re.getCause())) {
             return WebApplicationException.class.cast(re.getCause()).getResponse();
         }
+
+        if (ComponentException.class.isInstance(re)) {
+            ComponentException ce = (ComponentException) re;
+            throw new WebApplicationException(Response
+                    .status(ce.getErrorOrigin() == ComponentException.ErrorOrigin.USER ? 400 : 520, "Unexpected callback error")
```

Comment:
Let's map instead all cases  to :
- ErrorOrigin.USER -> 400
- ErrorOrigin.BACKEND -> 426
- ErrorOrigin.UNKNOWN -> 520

And let handle the 520 case by code below.

Reply:
done